File: backend/src/main/java/project/lms_rikkei_edu/modules/user/service/UserService.java
package project.lms_rikkei_edu.modules.user.service;

import project.lms_rikkei_edu.common.dto.response.PagedResponse;
import project.lms_rikkei_edu.modules.user.dto.request.AdminUserCreateRequest;
import project.lms_rikkei_edu.modules.user.dto.request.AdminUserListRequest;
import project.lms_rikkei_edu.modules.user.dto.request.AdminUserUpdateRequest;
import project.lms_rikkei_edu.modules.user.dto.request.ResetPasswordRequest;
import project.lms_rikkei_edu.modules.user.dto.response.AdminUserDetailResponse;
import project.lms_rikkei_edu.modules.user.dto.response.MessageResponse;
import project.lms_rikkei_edu.modules.user.dto.response.UserResponse;

import java.util.List;
import java.util.UUID;

public interface UserService {

    PagedResponse<UserResponse> getUsers(AdminUserListRequest request);

    AdminUserDetailResponse getUserDetail(UUID userId);

    UserResponse createUser(UUID adminId, AdminUserCreateRequest request);

    boolean existsByEmail(String email);

    boolean existsByPhoneNumber(String phoneNumber);

    UserResponse updateUser(UUID adminId, UUID userId, AdminUserUpdateRequest request);

    MessageResponse resetPassword(UUID adminId, UUID userId, ResetPasswordRequest request);

    List<UserResponse> batchCreateUsers(UUID adminId, List<AdminUserCreateRequest> requests, String courseTitle);
}

File: frontend/src/services/user-service.ts
import { httpClient } from '../lib';
import type { AdminUserCreateRequest, AdminUserDetailResponse, AdminUserListRequest, AdminUserUpdateRequest, MessageResponse, PagedResponse, UserResponse } from '../types';

export async function getUsers(params?: AdminUserListRequest) {
  const response = await httpClient.get<PagedResponse<UserResponse>>('/admin/users', { params });
  return response.data;
}

export async function getUserDetail(id: string) {
  const response = await httpClient.get<AdminUserDetailResponse>(`/admin/users/${id}`);
  return response.data;
}

export async function createUser(data: AdminUserCreateRequest) {
  const response = await httpClient.post<UserResponse>('/admin/users', data);
  return response.data;
}

export async function updateUser(id: string, data: AdminUserUpdateRequest) {
  const response = await httpClient.put<UserResponse>(`/admin/users/${id}`, data);
  return response.data;
}

export async function adminResetPassword(id: string, reason?: string) {
  const response = await httpClient.post<MessageResponse>(`/admin/users/${id}/reset-password`, { reason });
  return response.data;
}
